### PR TITLE
feat(release): #2123 default settlement

### PR DESCRIPTION
**Pull request by vgenev**
* Added config for renamed and separated central-settlement handlers
* Upgraded central-ledger and central-settlement to support default settlement functionality
* Added `"allow.auto.create.topics": true` to consumers configs
* Added  `"auto.create.topics.enable": true` to kafka broker config

### CODE REVIEW

File: mojaloop/values.yaml [hunk 1]
```diff
@@ -2376,19 +2376,18 @@ central:
         #  cpu: 100m
         #  memory: 128Mi
 
-    centralsettlement-handler-transfersettlement:
-      ## FOR_CGS
+    centralsettlement-handler-grosssettlement:
```
Comment (mdebarros) [inline mojaloop/values.yaml:2379]: `interchangeFeeCalculation.js` should not be defined for this service.

File: central/values.yaml [hunk 2]
```diff
@@ -2687,60 +2689,95 @@ centralsettlement:
...
-        #       multiply(transfer.amount.amount, 0.006, 2),
-        #       transfer.amount.currency,
-        #       payerFspId,
-        #       payeeFspId)
-        #   }
+        #   log('Running - DEFAULT EMPTY RULES SCRIPT')
+
+        ## Interchange Fee Calculation Script - Ensure that a settlementModel (refer to the associated tagged release on https://github.com/mojaloop/postman for an example) is created for the ledgerAccounType "INTERCHANGE_FEE".
+        ## Note: Uncomment this is you wish to enable CGS processing but also want Interchange Fee processing.
+        interchangeFeeCalculation.js: |
```
Comment (mdebarros) [inline central/values.yaml:2727]: Should this not be removed as its only applicable to the rules handler?

File: central/values.yaml [hunk 3]
```diff
@@ -2779,12 +2816,385 @@ centralsettlement:
...
+            name: http-api
+            externalPort: 3007
+    ingress:
+      enabled: false
       externalPath:
         api: /
       # Used to create an Ingress record.
       hosts:
-        api: central-settlement-transfersettlement.local
+        api: central-settlement-grosssettlement.local
```
Comment (mdebarros) [inline central/values.yaml:3197]: Is this Ingress not for the `centralsettlement-handler-rules`?

File: centralsettlement/values.yaml [hunk 4]
```diff
@@ -907,7 +911,380 @@ centralsettlement-handler-transfersettlement:
       api: /
     # Used to create an Ingress record.
     hosts:
-      api: central-settlement-transfersettlement.local
+      api: central-settlement-rules.local
```
Comment (mdebarros) [inline centralsettlement/values.yaml:914]: This should be for the `centralsettlement-handler-grosssettlement`?

File: centralsettlement/values.yaml [hunk 5]
```diff
@@ -563,7 +563,7 @@ centralsettlement-handler-settlementwindow:
     #  cpu: 100m
     #  memory: 128Mi
 
-centralsettlement-handler-transfersettlement:
+centralsettlement-handler-grosssettlement:
```
Comment (mdebarros) [inline centralsettlement/values.yaml:566]: `interchangeFeeCalculation.js` should not be defined this for the service.

File: centralsettlement/values.yaml [hunk 6]
```diff
@@ -907,7 +911,380 @@ centralsettlement-handler-transfersettlement:
...
+
+  ## metric configuration for prometheus instrumentation
+  metrics:
+    ## flag to enable/disable the metrics end-points
+    enabled: false
+    config:
+      timeout: 5000
+      prefix: moja_
+      defaultLabels:
+        serviceName: central-settlement-handler-grosssettlement
```
Comment (mdebarros) [inline centralsettlement/values.yaml:1035]: Rules handler?

File: centralsettlement/values.yaml [hunk 7]
```diff
@@ -907,7 +911,380 @@ centralsettlement-handler-transfersettlement:
...
+        api:
+          name: http-api
+          externalPort: 3007
+  ingress:
+    enabled: false
+    externalPath:
+      api: /
+    # Used to create an Ingress record.
+    hosts:
+      api: central-settlement-grosssettlement.local
```
Comment (mdebarros) [inline centralsettlement/values.yaml:1287]: Rules?

File: mojaloop/values.yaml [hunk 8]
```diff
@@ -2724,7 +2724,380 @@ central:
           api: /
         # Used to create an Ingress record.
         hosts:
-          api: central-settlement-transfersettlement.local
+          api: central-settlement-rules.local
```
Comment (mdebarros) [inline mojaloop/values.yaml:2727]: grosssettlement?

File: mojaloop/values.yaml [hunk 9]
```diff
@@ -2724,7 +2724,380 @@ central:
...
+
+      ## metric configuration for prometheus instrumentation
+      metrics:
+        ## flag to enable/disable the metrics end-points
+        enabled: false
+        config:
+          timeout: 5000
+          prefix: moja_
+          defaultLabels:
+            serviceName: central-settlement-handler-grosssettlement
```
Comment (mdebarros) [inline mojaloop/values.yaml:2848]: rules handler?

File: mojaloop/values.yaml [hunk 10]
```diff
@@ -2724,7 +2724,380 @@ central:
...
+            api:
+              name: http-api
+              externalPort: 3007
+      ingress:
+        enabled: false
+        externalPath:
+          api: /
+        # Used to create an Ingress record.
+        hosts:
+          api: central-settlement-grosssettlement.local
```
Comment (mdebarros) [inline mojaloop/values.yaml:3100]: Rules handler?

**Review by mdebarros — CHANGES_REQUESTED** [review]
Several comments for you attention.

File: centralledger/chart-handler-transfer-prepare/configs/default.json [hunk 11]
```diff
@@ -86,7 +86,6 @@
         ]
     },
     "INTERNAL_TRANSFER_VALIDITY_SECONDS": "432000",
-    "ENABLE_ON_US_TRANSFERS": {{ .Values.config.enable_on_us_transfers }},
```
[the remark this answers is not in the document]
Comment (vgenev) [reply centralledger/chart-handler-transfer-prepare/configs/default.json:89]: looks like some merge error

File: centralsettlement/chart-service/configs/default.json [hunk 12]
```diff
@@ -90,10 +96,11 @@
               "consumeTimeout": {{ .Values.config.kafka.consumer.notification.event.config.options.consumeTimeout }}
             },
             "rdkafkaConf": {
-              "client.id": "cs-con-transferfulfil-process",
-              "group.id": "cs-group-transferfulfil-process",
+              "client.id": "cs-con-notification-event",
```
[the remark this answers is not in the document]
Comment (vgenev) [reply centralsettlement/chart-service/configs/default.json:99]: but that's the notification-event consumer?

File: centralsettlement/requirements.yaml [hunk 13]
```diff
@@ -4,14 +4,14 @@ dependencies:
...
-# - name: centralsettlement-handler-transfersettlement
+#   repository: "file://./chart-handler-deferredsettlement"
+#   condition: centralsettlement-handler-deferredsettlement.enabled
+# - name: centralsettlement-handler-grosssettlement
 #   version: 12.0.0
-#   repository: "file://./chart-handler-transfersettlement"
-#   condition: centralsettlement-handler-transfersettlement.enabled
+#   repository: "file://./chart-handler-grosssettlement"
+#   condition: centralsettlement-handler-grosssettlement.enabled
 #- name: forensicloggingsidecar
```
[the remark this answers is not in the document]
Comment (vgenev) [reply centralsettlement/requirements.yaml:15]: yeah :(

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**